django/db/models/fields/__init__.py

```diff
@@ -1016,6 +1015,10 @@ def has_default(self):
         """Return a boolean of whether this field has a default value."""
         return self.default is not NOT_PROVIDED
 
+    def has_db_default(self):
```

Comment:
It's arguably bad practice to have a `has_` method returning a boolean instead of a `@property` as callers might be doing `field.has_default` (which will always be true) but as pointed in the comment message this was explicitly not made a `@property` to match how `has_default` is implemented and avoid and even messier situation.

We went through a deprecation for `User.is_authenticated()` and `.is_anonymous()` [in 1.10 so there's a known deprecation path](https://docs.djangoproject.com/en/5.1/releases/1.10/#using-user-is-authenticated-and-user-is-anonymous-as-methods) if we ever want to address this foot gun in the future but this is likely not the place to do so.

Reply:
Is `has_default` public API?

Reply:
> Is `has_default` public API?

That was only one Django docs search away; it's effectively not part of the `Field` documentation. It's nonetheless used by tons of third party applications that performs Django model fields introspection

- [DRF](https://github.com/search?q=repo%3Aencode%2Fdjango-rest-framework%20has_default&type=code)
- [Ninja](https://github.com/vitalik/django-ninja/blob/39f40b30ebcc368471a8c1449355990fa63ec0e2/ninja/orm/fields.py#L170)
- Most likely any third-party application database backends schema editor given the diff here.

So even if not part of the public API, if we were to consider changing them to `@property` (e.g. before actually making it part of the public API), we'd most likely need a deprecation period à la `CallableBool`.